File: CryptoApp/Shared/Labels/GFBodyLabel/GFBodyLabel.m
//
//  GFBodyLabel.m
//  CryptoApp
//
//  Created by Jansen Castillo on 1/7/25.
//

#import <Foundation/Foundation.h>
#import "GFBodyLabel.h"

@implementation GFBodyLabel

- (instancetype)init {
    self = [super initWithFrame:CGRectZero];
    if (self) {
        [self configure];
    }
    return self;
}

- (instancetype)initWithTextAlignment:(NSTextAlignment)textAlignment
                             fontSize:(CGFloat)fontSize
                               weight:(UIFontWeight)weight {
    self = [self init];
    if (self) {
        self.textAlignment = textAlignment;
        self.font = [UIFont systemFontOfSize:fontSize weight:weight];
    }
    return self;
}

- (void)configure {
    self.textColor = [UIColor labelColor];
    self.adjustsFontSizeToFitWidth = YES;
    self.adjustsFontForContentSizeCategory = YES;
    self.minimumScaleFactor = 0.8;
    self.numberOfLines = 1;
    self.translatesAutoresizingMaskIntoConstraints = NO;
}

@end

File: CryptoApp/Shared/Labels/GFBodyLabel.swift
//
//  CoinLabel.swift
//  CryptoApp
//
//  Created by Jansen Castillo on 25/6/25.
//
import UIKit

class GFBodyLabel: UILabel {

    override init(frame: CGRect) {
        super.init(frame: frame)
        configure()
    }

    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    convenience init(textAlignment: NSTextAlignment, fontSize: CGFloat = 14, weight: UIFont.Weight = .regular) {
        self.init(frame: .zero)
        self.textAlignment = textAlignment
        self.font = .systemFont(ofSize: fontSize, weight: weight)
    }


    private func configure() {
        textColor = .label
        adjustsFontSizeToFitWidth = true
        adjustsFontForContentSizeCategory = true
        minimumScaleFactor = 0.8
        numberOfLines = 1
        translatesAutoresizingMaskIntoConstraints = false
    }
}

